includes/abstracts/abstract.llms.database.query.php

```diff
@@ -556,4 +276,142 @@ protected function sql_orderby() {
+
+		list( $val, $has_replacement ) = $legacy_props[ $prop ];
+
+		$class     = get_called_class();
+		$is_method = method_exists( $this, $val );
+		$suffix    = $is_method ? '()' : '';
+		_deprecated_function( "Public access to property {$class}::{$prop}", '[version]', $has_replacement ? "{$class}::{$val}{$suffix}" : '' );
+
+	}
+
+	/**
+	 * Preserve backwards compat for read access to formerly public and removed class properties.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $key Property key name.
+	 * @return mixed
+	 */
+	public function __get( $key ) {
+
+		// Handle formerly public properties.
+		$legacy_props = $this->legacy_public_props();
+		if ( array_key_exists( $key, $legacy_props ) ) {
+			$this->public_prop_deprecation( $key );
+			$val = $legacy_props[ $key ][0];
+			return method_exists( $this, $val ) ? $this->$val() : $this->$val;
+		} elseif ( 'sql' === $key ) {
+			$class = get_called_class();
+			_deprecated_function( "Property {$class}::sql", '[version]', "{$class}::get_query()" );
+			return $this->query;
+		}
+
+	}
+
+	/**
+	 * Preserve backwards compat for write access to formerly public and removed class properties.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $key Property name.
+	 * @param mixed  $val Property value.
+	 * @return void
+	 */
+	public function __set( $key, $val ) {
+
+		$legacy_props = $this->legacy_public_props();
+		if ( array_key_exists( $key, $legacy_props ) ) {
+			$this->public_prop_deprecation( $key );
+			$this->$key = $val;
+		} elseif ( 'sql' === $key ) {
+			$class = get_called_class();
+			_deprecated_function( "Property {$class}::sql", '[version]', "{$class}::query" );
+			$this->query = $val;
+		}
+
+	}
+
+	// /**
+	// * Handle backwards compatibility for the misspelled (and removed) method `preprare_query()`.
+	// *
+	// * @since [version]
+	// *
+	// * @param string $name Method name.
+	// * @param array $args  Arguments passed to the method.
+	// * @return void|string
+	// */
+	// public function __call( $name, $args ) {
+	// if ( 'preprare_query' === $name ) {
+	// $class = get_called_class();
+	// _deprecated_function( "{$class}::preprare_query()", '[version]', "{$class}::prepare_query()" );
+	// return $this->prepare_query();
+	// }
+	// }
+
+	/**
+	 * Prepare the query.
+	 *
+	 * Should return the query which will be used by `query()`.
+	 *
+	 * This *should* be an abstract method but is defined here for backwards compatibility
+	 * to preserve the previous method, `preprare_query()` (notice the misspelling).
+	 *
+	 * Once the `preprare_query()` method is fully removed in the next major release this
+	 * method can be removed in favor of the abstract from the parent class.
+	 *
+	 * @since [version]
+	 *
+	 * @return mixed
+	 */
+	protected function prepare_query() {
+		if ( method_exists( $this, 'preprare_query' ) ) {
+			$class = get_called_class();
+			// _deprecated_function( "{$class}::preprare_query()", '[version]', "{$class}::prepare_query()" );
```

Comment:
```suggestion
```
right?

Reply:
I was arguing with myself about whether to update the whole codebase. I commented it out to test things without the warning throwing so I could see if it was working properly. I'm going to restore it later.